app/models/holding.rb

```diff
@@ -1,4 +1,19 @@
 class Holding < ApplicationRecord
     belongs_to :user
     belongs_to :company
+
+    def average_cost
+        unless @average_cost
+            # Use FIFO in averaging buys, ignoring first N sells
+            num_sells = Trade.where(company_id: self.company_id).joins(:sell_order).where('orders.user_id' => self.user_id).count
+
+            # Successful buys
+            fifo_buy_total = Trade.where(company_id: self.company_id).joins(:buy_order).where('orders.user_id' => self.user_id).order('id DESC').limit(9223372036854775807).offset(num_sells).inject(0) { |sum, x|
+                sum + x.price
+            }
+
+            @average_cost = fifo_buy_total / self.quantity
```

Comment:
500 Error: ActionView::Template::Error (divided by 0)

Reply:
Just made #24 for this, thanks for catching!